crates/ruma-macros/src/identifiers.rs

```diff
@@ -655,6 +622,35 @@ fn expand_unchecked_impls(input: &ItemStruct) -> TokenStream {
     }
 }
 
+fn expand_as_str_impls(ty: TokenStream, impl_generics: &ImplGenerics<'_>) -> TokenStream {
+    let partial_eq_string = expand_partial_eq_string(ty.clone(), impl_generics);
```

Comment:
Is `expand_partial_eq_string` also used elsewhere? If not, might make sense to inline.

Reply:
Yes, it is used separately for `Box<#ty>`